plugins/ffmpeg_plugin/kahawai.c

```diff
@@ -0,0 +1,574 @@
+/*
+ * Kahawai raw video demuxer
+ * Copyright (c) 2022 Intel
+ *
+ * This file is part of FFmpeg.
+ *
+ * FFmpeg is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * FFmpeg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with FFmpeg; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "libavutil/imgutils.h"
+#include "libavutil/parseutils.h"
+#include "libavutil/pixdesc.h"
+#include "libavutil/opt.h"
+#include "libavformat/avformat.h"
+#include "libavformat/internal.h"
+
+#include <st_pipeline_api.h>
+#include <st_convert_api.h>
+
+#define KAHAWAI_FRAME_BUFFER_COUNT 8
```

Comment:
We should consider move this to a runtime value in future enhancement, maybe use a option like "-fb_cnt"

Reply:
Modified.